pkg/tagger/global.go

```diff
@@ -17,38 +17,58 @@ import (
 var defaultTagger *Tagger
 var initOnce sync.Once
 
-// fullCardinality caches the value of the full_cardinality_taggin option
-var fullCardinality bool
+// checksFullCardinality holds the config that says whether we should send high cardinality tags for checks metrics
+// this can still be overridden when calling get_tags in python checks.
+var checksFullCardinality bool
+
+// dogstatsdFullCardinality holds the config that says whether we should send high cardinality tags for metrics from
+// dogstatsd.
+var dogstatsdFullCardinality bool
 
 // Init must be called once config is available, call it in your cmd
 // defaultTagger.Init cannot fail for now, keeping the `error` for API stability
 func Init() error {
 	initOnce.Do(func() {
-		fullCardinality = config.Datadog.GetBool("full_cardinality_tagging")
+		checksFullCardinality = config.Datadog.GetBool("send_container_tags_for_checks")
+		dogstatsdFullCardinality = config.Datadog.GetBool("send_container_tags_for_dogstatsd")
 		defaultTagger.Init(collectors.DefaultCatalog)
 	})
 	return nil
 }
 
-// Tag queries the defaultTagger to get entity tags from cache or sources
+// Tag queries the defaultTagger to get entity tags from cache or sources.
+// It can return tags at high cardinality (with tags about individual containers),
+// or at orchestrator cardinality (pod/task level)
 func Tag(entity string, highCard bool) ([]string, error) {
-	return defaultTagger.Tag(entity, highCard)
+	if highCard == true {
+		return defaultTagger.Tag(entity, collectors.HighCardinality)
+	}
+	return defaultTagger.Tag(entity, collectors.OrchestratorCardinality)
```

Comment:
Does this mean calling `tagger.Tag(entity, false)` will now return pod-level cardinality? This would impact dogstatsd, docker/kube.containers.running gauges and AD with no other option than to downgrade if something goes wrong.

Reply:
@xvello yeah I wrote that a few hours ago and have been thinking on it since then. I'm not happy about it, and so far I see two ways to improve it:

- add a `DefaultCardinality` setting which value would replace `collectors.OrchestratorCardinality` in this line
- change the signature of `tagger.Tag` to take a cardinality instead of a `bool` and update all calls to it (there's not too many afaict)

Do you think of any other solution? If not, which one of these do you prefer?

Reply:
💯 agreed, updating.